tests/system_tests_panic_handler.py

```diff
@@ -0,0 +1,113 @@
+
+    @classmethod
+    def setUpClass(cls):
+        super(PanicHandlerTest, cls).setUpClass()
+
+        config = [
+            ('router', {'mode': 'interior', 'id': "PanicRouter"}),
+            ('listener', {'role': 'normal',
+                          'port': cls.tester.get_port()}),
+            ('address', {'prefix': 'closest', 'distribution': 'closest'}),
+            ('address', {'prefix': 'multicast', 'distribution': 'multicast'}),
+        ]
+
+        # turn off python debug for this test to prevent it from dumping stuff
+        # to output
+        os.environ.pop("PYTHONMALLOC", None)
+        os.environ.pop("PYTHONDEVMODE", None)
+        config = Qdrouterd.Config(config)
+        cls.router = cls.tester.qdrouterd("PanicRouter", config, wait=True,
+                                          cl_args=["-T"], expect=Process.EXIT_FAIL)
+
+    def test_01_panic_handler(self):
+        """
+        Crash the router and verify the panic handler writes the crash to
+        stderr
+        """
+        self.router.wait_startup_message()
+
+        mgmt = QdManager(address=self.router.addresses[0], timeout=1)
+        # this call will crash the router
+        try:
+            mgmt.create('io.skupper.router.httpListener',
+                        {'address': 'closest/panic',
+                         'port': self.tester.get_port(),
+                         'protocolVersion': 'HTTP1',
+                         'host': '$FORCE$CRASH$SEGV$'})
+            self.assertTrue(False, 'mgmt.create() should have crashed')
+        except Exception as exc:
+            pass  # expected - the router just crashed
+
+        self.assertTrue(retry(lambda: self.router.poll() is not None))
+
+        # do some simple validation on the generated panic output:
+
+        crash_dump = ""
+        with open(self.router.outfile_path, 'r') as fd:
+            crash_dump = fd.read()
+
+        regex = re.compile(r"^\*\*\* SKUPPER-ROUTER FATAL ERROR \*\*\*", re.MULTILINE)
+        match = regex.search(crash_dump)
+        self.assertIsNotNone(match, "failed to match banner")
+
+        regex = re.compile(r"^Version: (\S+)$", re.MULTILINE)
+        match = regex.search(crash_dump)
+        self.assertIsNotNone(match, "failed to find Version:")
+
+        regex = re.compile(r"^Signal: (\d+) (\S+)$", re.MULTILINE)
+        match = regex.search(crash_dump)
+        self.assertIsNotNone(match, "failed to find Signal:")
+
+        regex = re.compile(r"^Process ID: (\d+) (\S+)$", re.MULTILINE)
+        match = regex.search(crash_dump)
+        self.assertIsNotNone(match, "failed to find Process ID:")
+
+        regex = re.compile(r"^!!! libunwind not present: backtrace unavailable !!!$",
+                           re.MULTILINE)
+        if (regex.search(crash_dump) is None):
+            # libunwind present, expect at least one stack frame:
+            regex = re.compile(r"^\[(\d+)\] IP: ", re.MULTILINE)
+            self.assertTrue(len(regex.findall(crash_dump)) > 0,
```

Comment:
this could be `self.assertGreater`, just so that failure message is nicer

Reply:
or assertRegex after all, https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRegex

Reply:
changed to assertRegex